### DISCUSSION TITLE
ASGI server vs HTTP server requests

### DISCUSSION THREAD

**Discussion by aaronamelgar** [fastapi/fastapi#3178] [Questions] [answered]
Hi there, has anyone used Sentry for performance monitoring? I'm just starting a project and am seeing a couple different types of requests come through: Generic ASGI request and router requests, hitting both ASGI.server and HTTP.server operations (see image attached). The ASGI.server request seems to last several minutes, perhaps only on 503 uncaught exceptions but I'm not sure yet. I'm using Heroku for now with profile set below, just for this MVP until moving into AWS. All of my routes are async routes 

web: gunicorn -k uvicorn.workers.UvicornWorker api:api

1) Why are some requests asgi while others are http? the majority of them say http.server including the simple health check code below. Is this just a Sentry specific logging system?

```
@api.get("/health")
async def health_check(db=Depends(get_async_session)):
    r = await db.execute('SELECT 1')
    return {"status": "healthy"}
```

2) Why do some of asgi.server requests run for so long? I'm guessing on uncaught exceptions some event loop isn't closing

<img width="749" alt="Screen Shot 2021-05-05 at 7 17 05 AM" src="https://user-images.githubusercontent.com/42826952/117133177-e7650880-ad71-11eb-949d-1b3b0374fe82.png">

**Comment by tiangolo** [comment, marked as answer]
Hey there, Sentry now has a FastAPI integration, that might solve your use case: https://docs.sentry.io/platforms/python/guides/fastapi/

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
